packages/plugin-core/src/components/lookup/vaultSelectionModeKeeper.ts

```diff
@@ -0,0 +1,101 @@
+import { VaultSelectionMode } from "./types";
+import { getDWorkspace } from "../../workspace";
+import { ConfigUtils } from "@dendronhq/common-all";
+import * as _ from "lodash";
+
+/**
+ * Class responsible for keeping track of the desired mode for vault
+ * selection.
+ *
+ * We want to start with the selection from configuration value
+ * However, if the user has toggled the selection during the session
+ * we want to at least respect their wishes during the session and use
+ * the value that they have toggled to during the session.
+ *
+ * (V2: in the future we should highly consider modifying the configuration
+ * value if the user has changed the toggle to persist user's wishes across
+ * the sessions without users having to modify the config yaml).
+ * */
```

Comment:
Some thoughts on the keeper class in general:

Right now, the user experience we provide with configuration is less than optimal (having to open a yaml file to modify it), but this I think is something we should address separately.

I think keeping the user's expectations simple is beneficial for both the users and the developers: 
1. users are happy that there are no surprises (when they toggle the vault selection mode that deviates from the config, are they notified that this mode will persist for this session? and that it will reset when a new session starts?), and 
2. developers have less moving parts to be concerned (especially for a common component that is shared across many features of Dendron)

Right now, the immediate expectation mismatch I'm seeing is that the user will not know that these modes are going to be remembered for the session.

I definitely think this could provide a smoother experience if we introduce this too keeper behavior to other configs as well, but currently there is no guardrails against the users thinking the other togglable configs will be remembered as well. (Not suggesting we should implement this for everything right now)

If we decided that we are going with this solution, I think we should properly notify the users that whatever they toggle right now is going to be remembered for this session and will be reset to the config value when they start a new one.

That being said, maybe it is worth introducing this to the users and gather feedback on how it feels. If the users like this, we can generalize the keeper component so that we can reuse it in other places that makes sense.

Any thoughts on this @kevinslin ?

Reply:
I agree with @hikchoi - lets add the ability to make this selection sticky in future work

Reply:
>  lets add the ability to make this selection sticky in future work

Could we clarify whether we want to:
1. Add a notification to the user that they have toggled the value and that the value will persist for the duration of this session. 
2. Not persist the value for duration of the session.